xdsl/ir.py

```diff
@@ -204,9 +204,13 @@ def owner(self) -> Operation:
         return self.op
 
     def __repr__(self) -> str:
-        return f"OpResult(typ={repr(self.typ)}, num_uses={repr(len(self.uses))}, " + \
-               f"op_name={repr(self.op.name)}, " + \
-               f"result_index={repr(self.result_index)}, name={repr(self.name)})"
+        return "{}<%{}: {}>(num_uses={}, op_name={})".format(
```

Comment:
I would write it `{}(name={}, type={}, result_index={}, num_uses={}, op_name={})` instead.
Having this `<%"name": type>`  is a bit confusing, especially since names wouldn't stay the same once you show it in a large program, and also since you don't see the `result_index` when the value has a name.